Uplift/Models/Gym.swift

```diff
@@ -135,136 +126,189 @@ struct DailyGymHours {
+    var timeRanges: [FacilityHoursRange] = []
+
+    init(facilityHoursData: AllGymsQuery.Data.Gym.Facility.Detail.Time) {
+        dayOfWeek = facilityHoursData.day
+        timeRanges = facilityHoursData.timeRanges.compactMap({ rangeData -> FacilityHoursRange? in
+            guard let rangeData = rangeData else { return nil }
+            return FacilityHoursRange(facilityHoursRangeData: rangeData)
+        })
     }
 
-    init(facilityHoursData: GymByIdQuery.Data.Gym.Facility.Time?) {
-        if let facilityHours = facilityHoursData {
-            dayOfWeek = facilityHours.day
-            openTime = Date.getTimeFromString(datetime: facilityHours.startTime)
-            closeTime = Date.getTimeFromString(datetime: facilityHours.endTime)
-        } else {
-            openTime = Date()
-            closeTime = openTime
-            dayOfWeek = 0
-        }
+    init(facilityHoursData: GymByIdQuery.Data.Gym.Facility.Detail.Time) {
+        dayOfWeek = facilityHoursData.day
+        timeRanges = facilityHoursData.timeRanges.compactMap({ rangeData -> FacilityHoursRange? in
+            guard let rangeData = rangeData else { return nil }
+            return FacilityHoursRange(facilityHoursRangeData: rangeData)
+        })
     }
-    
 }
 
-struct Facility {
+struct FacilityHoursRange {
+    var openTime: Date = Date()
+    var closeTime: Date = Date()
+    var specialHours: Bool = false
+    var restrictions: String = ""
+
+    init(facilityHoursRangeData: AllGymsQuery.Data.Gym.Facility.Detail.Time.TimeRange) {
+        openTime = Date.getTimeFromString(datetime: facilityHoursRangeData.startTime)
+        closeTime = Date.getTimeFromString(datetime: facilityHoursRangeData.endTime)
+        specialHours = facilityHoursRangeData.specialHours
+        restrictions = facilityHoursRangeData.restrictions
+    }
 
-    var equipment: [Equipment]
-    var miscInformation: [String]
-    var name: String
-    var times: [DailyGymHours]
+    init(facilityHoursRangeData: GymByIdQuery.Data.Gym.Facility.Detail.Time.TimeRange) {
+        openTime = Date.getTimeFromString(datetime: facilityHoursRangeData.startTime)
+        closeTime = Date.getTimeFromString(datetime: facilityHoursRangeData.endTime)
+        specialHours = facilityHoursRangeData.specialHours
+        restrictions = facilityHoursRangeData.restrictions
+    }
+}
 
-    init(facilityData: AllGymsQuery.Data.Gym.Facility) {
-        if let equipmentGymData = facilityData.equipment {
-            equipment = equipmentGymData.compactMap({ equipmentData -> Equipment? in
+enum DetailType: String {
+    case equipment = "Equipment"
+    case hours = "Hours"
+    case prices = "Prices"
+    case subfacilities = "Sub-Facilities" // MISCELLANEOUS
+}
+
+struct FacilityDetail {
+
+    var detailType: DetailType
+
+    // Equipment
+    var equipment: [Equipment] = []
+
+    // Hours
+    var times: [DailyFacilityHoursRanges] = []
+
+    // Prices
+    var items: [String] = []
+    var prices: [String] = []
```

Comment:
same here

Reply:
depending on case most of these will be empty in the response, is is better to just default them to empty or assign them within each case

Reply:
or remove the switch statement and iterate over every field for every case?

Reply:
oh ok i see. we can keep the default values then